app/src/main/java/org/listenbrainz/android/presentation/features/yim/YimNetworkViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package org.listenbrainz.android.presentation.features.yim
+
+import android.content.Context
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.listenbrainz.android.util.connectivityobserver.ConnectivityObserver
+import org.listenbrainz.android.util.connectivityobserver.NetworkConnectivityObserver
+
+class YimNetworkViewModel (context: Context) : ViewModel() {
```

Comment:
Make this generic because this can be used everywhere

Reply:
Great idea! Should i add this to util package?